### PR TITLE
Fix conversationhandler

### LINKED ISSUE #1031: Repeating job AttributeError
Hi!

I'm having a small def which is a repeating job, the code actually looks like this:
```
dp = updater.dispatcher
j = updater.job_queue
j.run_repeating(rss, 300)
```

After upgrading to 10.0.0 I get an exception in logs.

### Configuration
**Operating System:** Ubuntu

**Version of Python, python-telegram-bot & dependencies:**

python-telegram-bot 10.0.0
certifi 2018.01.18
future 0.16.0
Python 3.6.3 (default, Oct  6 2017, 08:44:35)

### Logs
```
AttributeError: 'NoneType' object has no attribute 'schedule_removal'
[03-03 23:51:50] [ERROR] [bot]
An uncaught error was raised while processing the update
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/telegram/ext/dispatcher.py", line 279, in process_update
    handler.handle_update(update, self)
  File "/usr/local/lib/python3.6/dist-packages/telegram/ext/conversationhandler.py", line 309, in handle_update
    timeout_job.schedule_removal()
AttributeError: 'NoneType' object has no attribute 'schedule_removal'
```

### PR DESCRIPTION
**Pull request by Eldinnie**
As found by @nmlorg and described in #1031
Went for the easiest fix.

fixes #1031

### CODE REVIEW

**Comment by Eldinnie** [comment]
@Eldinnie From @nmlorg

I think it would be something like (with conversation_timeout=10):

t=0 user sends /start
new_state is THIRSTY
bot checks self.timeout_jobs[key] but it is None
bot sets self.timeout_jobs[key] = CANCEL-JOB-1

t=5 user sends /brew
new_state is BREWING
bot checks self.timeout_jobs[key] and it is CANCEL-JOB-1 (not None), but new_state is not END, so it ignores it
bot overwrites self.timeout_jobs[key] = CANCEL-JOB-2

t=10 CANCEL-JOB-1 runs

t=12 user sends /pourCoffee, but the conversation has been canceled

t=15 CANCEL-JOB-2 runs

**Comment by nmlorg** [comment]
Uh oh, I originally thought this wasn't super high priority because conversation_timeout was new and it would only trigger if someone changed their bot to use it, but that's not actually the case. The existing test_end_on_first_message test almost exposes the problem, but the relevant exception is suppressed by [Dispatcher.process_update's exception catch-all](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/ext/dispatcher.py#L299). I'm pretty sure this actually affects all current users of ConversationHandler whenever an entry state transitions directly to END or whenever any state transitions to END without having conversation_timeout set.

File: telegram/ext/conversationhandler.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def handle_update(self, update, dispatcher):
         new_state = self.current_handler.handle_update(update, dispatcher)
         timeout_job = self.timeout_jobs.get(self.current_conversation)
 
-        if timeout_job is not None or new_state == self.END:
+        if timeout_job is not None and new_state == self.END:
```
Comment (nmlorg) [inline telegram/ext/conversationhandler.py]: So with this logic, this new test:
```python
   def test_conversation_timeout_keeps_extending(self, dp, bot, user1):
        handler = ConversationHandler(entry_points=self.entry_points, states=self.states,
                                      fallbacks=self.fallbacks, conversation_timeout=0.5)
        dp.add_handler(handler)

        # Start state machine, wait, do something, verify the timeout is extended.
        # t=0 /start (timeout=.5)
        # t=.25 /brew (timeout=.75)
        # t=.5 original timeout
        # t=.6 /pourCoffee (timeout=1.1)
        # t=.75 second timeout
        # t=1.1 actual timeout
        message = Message(0, user1, None, self.group, text='/start', bot=bot)
        dp.process_update(Update(update_id=0, message=message))
        assert handler.conversations.get((self.group.id, user1.id)) == self.THIRSTY
        sleep(0.25)  # t=.25
        dp.job_queue.tick()
        assert handler.conversations.get((self.group.id, user1.id)) == self.THIRSTY
        message.text = '/brew'
        dp.process_update(Update(update_id=0, message=message))
        assert handler.conversations.get((self.group.id, user1.id)) == self.BREWING
        sleep(0.35)  # t=.6
        dp.job_queue.tick()
        assert handler.conversations.get((self.group.id, user1.id)) == self.BREWING
        message.text = '/pourCoffee'
        dp.process_update(Update(update_id=0, message=message))
        assert handler.conversations.get((self.group.id, user1.id)) == self.DRINKING
        sleep(.4)  # t=1
        dp.job_queue.tick()
        assert handler.conversations.get((self.group.id, user1.id)) == self.DRINKING
        sleep(.1)  # t=1.1
        dp.job_queue.tick()
        assert handler.conversations.get((self.group.id, user1.id)) is None
```
fails at the t=.6 test:
```
        assert handler.conversations.get((self.group.id, user1.id)) == self.BREWING
        sleep(0.35)  # t=.6
        dp.job_queue.tick()
>       assert handler.conversations.get((self.group.id, user1.id)) == self.BREWING
E       assert None == 1
```
However, if this is changed to just `if timeout_job is not None:`, everything's happy.

**Comment by alexbagirov** [comment]
Hello again, will try it as soon as I can.

**Comment by Eldinnie** [comment]
What he could do is clone the library, checkout the branch and pip install from the cloned env.

**Pull request merged**